combat-trainer.lic

```diff
@@ -1425,15 +1426,55 @@ class AbilityProcess
     @barb_buffs = @buffs.delete('barb_buffs') || []
     echo("  @barb_buffs: #{@barb_buffs}") if $debug_mode_ct
     setup_barb_buff_flags
+
+    @paladin_use_badge = settings.paladin_use_badge
+    echo("  @kneel_khri: #{@kneel_khri}") if $debug_mode_ct
+
+    return unless @paladin_use_badge
+
+    if DRCI.wearing?('pilgrim badge')
+      @paladin_badge_get_verb = 'remove'
+      @paladin_badge_store_verb = 'wear'
+    else
+      @paladin_badge_get_verb = 'get'
+      @paladin_badge_store_verb = 'stow'
+    end
+
+    UserVars.paladin_last_badge_use ||= Time.now
+    @badge_reuse_cooldown = 1860 # 31 minutes
```

Comment:
I wonder if this should be in TrainerProcess... then you get the timer for free. It wouldn't be based on a skill check, but we have a precedent for that (FavorOrb: https://github.com/jayRyan24/dr-scripts/blob/60839aab83cc6b9829b9ce4f37bd870174c5bbb7/combat-trainer.lic#L1580)

Reply:
Is this used to train a skill? I thought it was used to maintain high levels of soul.

Reply:
oh sorry, misread. I want the overall timer so we can use it during crossing-training too.  Think I left a comment to this effect in the main comments section previous.